### ISSUE TITLE
Prettier/Cosmiconfig – Leaky process.argv throws error

### ISSUE THREAD

**Issue by amiller-gh** [prettier/prettier#2797]
**Prettier Version:** 1.6.1 / master

**Code**

`index.js`
```js
import * as prettier from "prettier";

prettier.format('let foo: boolean = true', {
  parser: 'typescript'
});
```
*handwave-y magic* `index.js` is imported in a webpack plugin

Script is run:
```bash
webpack --config webpack-config/webpack.chrome.js --config webpack-config/webpack.ie.js
```

**Expected behavior:**
Prettier runs as expected, returns string with formatted code.

**Actual behavior:**
```
TypeError: Path must be a string. Received [ 'webpack-config/webpack.chrome.js', 'webpack-config/webpack.ie.js' ]
```
We are trying to use prettier programmatically as part of a webpack plugin. Webpack allows multiple `--config` options to be passed in. On initialization of `cosmiconfig` (https://github.com/prettier/prettier/blob/master/src/resolve-config.js#L6) the library looks at `process.argv` for a `--config` cli option (https://github.com/davidtheclark/cosmiconfig/blob/master/index.js#L30). It picks up the `--config` value is intended for webpack and throws when its not a simple string. 

When used programmatically, these `cosmiconfig` objects should never be used and don't need to be called unless `resolveConfig` or `resolveConfigSync` are explicitly called. However, because `src/resolve-config` is being included in index (https://github.com/prettier/prettier/blob/master/index.js#L12), and the `cosmiconfig` calls are at the root level in the module, `cosmiconfig` is being called even when `resolveConfig` is never used.

**Comment by azz** [comment]
You can pass `--no-config` to prettier to get around this.

**Comment by azz** [comment]
Sorry, missed that your were using the API, not CLI

**Comment by amiller-gh** [comment]
Paper-trail: Closing PR on `prettier` in favor of a fix in `cosmiconfig`. If/when pulled in and released should just require a version bump to resolve.

**Comment by amiller-gh** [comment]
Hey @azz (sorry to be a bother!!) but this appears to be fixed in master, do you have an ETA on next release? My product is blocked until this lands.

**Comment by ikatyang** [comment]
Closing as it's already fixed by #2843.
